### ISSUE TITLE
invalidate() and refresh

### ISSUE THREAD

**Issue by mbfakourii** [Ahmed-Abdelmeged/PagingLibraryWithRxJava#4]
Hello when i use the invalidate() and RecyclerView is on the page for example 7 then the invalidate() is scrolled to page 2

```kotlin
 sourceFactory.usersDataSourceLiveData.value!!.invalidate()
```

**Comment by eternalBlast** [comment]
i got this issue too. Have you fixed this issue?

**Comment by mbfakourii** [comment]
No, I haven't got anything yet but I realized this is a problem on the part of Google

**Comment by eternalBlast** [comment]
I think so. I tried to get the key from `adapter.currentList.lastKey` but always return null. I am trying another ways.

**Comment by eternalBlast** [comment]
Are you using PagedKeyedDataSource or ItemKeyedDataSource?

**Comment by mbfakourii** [comment]
No, I didn't use it!
Did you solve the problem?

**Comment by eternalBlast** [comment]
I solved use some hacky solution :)
So, whenever I observed the network state, I scroll the recyclerview to top if the swipe refresh is in refreshing state. Here is the snippet:
```
yourViewModel.getNetworkState().observe(
            viewLifecycleOwner,
            androidx.lifecycle.Observer<NetworkState> {
                    if (bd.swipeRefreshLayout.isRefreshing)
                        bd.historyRV.smoothScrollToPosition(0)
                    bd.swipeRefreshLayout.isRefreshing = false
})
```
Please share if you have better solution.

**Comment by mbfakourii** [comment]
> I solved use some hacky solution :)
> So, whenever I observed the network state, I scroll the recyclerview to top if the swipe refresh is in refreshing state. Here is the snippet:
> 
> ```
> yourViewModel.getNetworkState().observe(
>             viewLifecycleOwner,
>             androidx.lifecycle.Observer<NetworkState> {
>                     if (bd.swipeRefreshLayout.isRefreshing)
>                         bd.historyRV.smoothScrollToPosition(0)
>                     bd.swipeRefreshLayout.isRefreshing = false
> })
> ```
> 
> Please share if you have better solution.

I don't think it's a good way

**Comment by eternalBlast** [comment]
This solution is for temporary. Can you provide cleaner solution?
